ITE/11/README.adoc

```diff
@@ -0,0 +1,285 @@
+that the builder made no network calls. But first, it is necessary to verify
+that the monitor was of an expected type and that its trace policy included
+tracking network calls.
+
+```yaml
+...
+expectedAttributes:
+    - "predicate.monitor.type == 'https://github.com/cilium/tetragon'"
+    - "predicate.monitor.tracePolicy.policies.exists(p, p['Name'] == 'connect')"
+    - "size(predicate.monitorLog.network) == 0"
+...
+```
+
+=== Example: Verifying Contents of Opaque Fields in in-toto Links
+
+As noted above, in-toto links have an opaque `byproducts` field that records
+additional information pertaining to the step performed. in-toto's verification
+workflow does not verify its contents by default. However, the changes proposed
+here can be used to add checks for its contents as well as those of the other
+opaque field, `environment`.
+
+```yaml
+...
+expectedAttributes:
+    - "size(predicate.byproducts['stderr']) == 0"
+    - "predicate.byproducts['return-value'] == 0"
+    - "predicate.environment['<key>'] == '<expected value>'"
+...
+```
+
+[[reasoning]]
+== Reasoning
+
+This ITE presents certain changes that necessitate further reasoning.
+
+=== Use of CEL
+
+CEL is a lightweight language designed for expression evaluations. One of its
+key applications is for security policies, making it a good fit for specifying
+attribute rules in in-toto layouts. CEL expressions are evaluated in linear time
+and the language is not Turing-complete, mitigating concerns about arbitrary
+executions during verification. Also, it may be possible to generate CEL rules
+from other higher-level languages to avoid having developers write CEL
+expressions manually.
+
+Finally, another feature making CEL a good fit for in-toto layout attribute
+checks is that it was designed to be embedded in other applications. Indeed,
+in-toto is not the first system to support embedded CEL expressions--Kubernetes
+also supports using CEL to declare various constraints as does Envoy in its Role
+Based Access Control policies.
+
+=== Supporting Other Languages
+
+Other policy specific languages such as Open Policy Agent's Rego and CUE are
+alternatives to using CEL. Also, other expression languages with similar
+capabilities as CEL may also be preferred by some users. Supporting these
+choices (and identifying the language used for a rule) is an open discussion
+point. CEL is simple to write and embed into in-toto layouts but is also less
+powerful than full-fledged policy languages. That said, the focus is on setting
+constraints on attributes recorded in in-toto attestations, which CEL is capable
+enough for. This may change if use cases emerge that CEL cannot express
+constraints for. It's also worth noting that for niche scenarios, the original
+method of writing custom inspections also remains an option.
+
+[[backwards-compatibility]]
+== Backwards Compatibility
+
+This ITE is entirely backwards compatible with both the in-toto v1.0 layout
```

Comment:
Backwards compatible with what?  Is this a legacy client and a new layout?  A new client and a legacy layout, etc.?  I don't think it is always backwards compatible.

Reply:
Clarified this some more to indicate that backwards compatibility isn't a given, we may have situations where a check that ought to have failed actually passes because a legacy client skips attribute checks.